### ISSUE TITLE
Handle groupmemberships over SAML/JWT by following graph URI

### ISSUE THREAD

**Issue by jmartens** [snok/django-auth-adfs#174]
When a large number of memberships is supplied in the auth token the individual groups are not provided anymore. Instead a Graph URI is provided. By following that you should be able to retrieve all group memberships for the user.

AFAICT the [limit is 150 for SAML requests and 200 for JWT requests](https://docs.microsoft.com/en-us/azure/active-directory/hybrid/how-to-connect-fed-group-claims).

A workaround on the Azure side is to configure your application for specific groups only and set the groups claim to only return groups assigned to this application ([as can be seen in the screenshot here](https://github.com/snok/django-auth-adfs/issues/173#issuecomment-896917356)), but it would be nice if this could be handled by this framework as well by following the Graph URI provided and parsing its results.

**Comment by JonasKs** [comment]
Hi, this seems like a good suggestion indeed.   
PR welcome. :blush:

**Comment by jmartens** [comment]
Not sure if I can find the time, but will see what I can do. Already have a hunch on where to add it in the code, not sure on the implementation details though.

Any preference on how to run rest requests and process the results? Are there helpers or classes in this package you prefer to be used?

**Comment by jmartens** [comment]
> Already have a hunch on where to add it in the code

I am guessing an additional leaf in this if else clause is the proper place to check for a d retrieve the Graph Api url:
https://github.com/snok/django-auth-adfs/blob/4917d379153beeb2d9942a9b938e3b75816b1266/django_auth_adfs/backend.py#L195

**Comment by JonasKs** [comment]
Yeah, I'd say there too. The graph API has to be optional, though. 😊 It'll add some overhead, so should be opt in.

**Comment by JonasKs** [comment]
I've never worked on a Azure AD Django project, so i'm not you man to implement this. 

Currently writing an Azure AD plug-in for FastAPI, but it won't be as comprehensive as this package.

**Comment by jmartens** [comment]
> It'll add some overhead, so should be opt in.

OK, will see if and how I can create a settings parameter for it.

**Comment by jmartens** [comment]
[This SO post](https://stackoverflow.com/questions/57115945/groups-overage-claim) might also hold some vital clues to get it working as apparently the application needs for `Directory.Read.All` permissions to successfully request the group memberships using the returned Graph URI.

**Comment by jmartens** [comment]
To get the URL for the claim I think this should work:

```(python=
try:
  claim_name = claims['_claim_names'][settings.GROUPS_CLAIM]
  claim_source = claims['_claim_sources'][claim_name]
  claim_url = claim_source['endpoint']
except KeyError as e:
  logger.error(f"Key not found: {e}")
```

**Comment by stefanoostwegel** [comment]
Has anyone got an update regarding this issue?
Would love to see the graph integration work as we have far to many users with to many groups.
Now we have to administer azure AD and Django, in stead of only django.

**Comment by JonasKs** [comment]
Not much update to give, other than a PR is welcome.
